src/js/components/DataTable/Resizer.js

```diff
@@ -8,40 +8,48 @@ import React, {
 import styled from 'styled-components';
 
 import { Box } from '../Box';
-import { Button } from '../Button';
 import { Keyboard } from '../Keyboard';
-import { Stack } from '../Stack';
 import { useThemeValue } from '../../utils/useThemeValue';
 import { MessageContext } from '../../contexts/MessageContext';
+import { focusStyle, unfocusStyle } from '../../utils/styles';
 
-// Added a temporary min-width of 2px here so that the element doesn't
-// end up with a width of 0px. This is a placeholder solution until we
-// revisit this in https://github.com/grommet/grommet/issues/7273
-const InteractionBox = styled(Button)`
-  min-width: 2px;
+const StyledResizer = styled(Box)`
+  position: absolute;
+  right: 0;
+  width: 24px;
+  height: 100%;
+  top: 0;
   cursor: col-resize;
-  > * {
-    opacity: 0;
+  z-index: 1;
+  &:focus {
+    ${(props) =>
+      (!props.plain || props.focusIndicator) &&
+      focusStyle({ inset: props.focusIndicator === 'inset' })}
   }
-
-  // when mouse down, we want to continue to display styling
-  ${(props) => props.active && '> * { opacity: 1; }'}
-
-  &:hover {
-    > * {
-      opacity: 1;
-    }
+  &:focus:not(:focus-visible) {
+    ${unfocusStyle()}
   }
 `;
 
-const Resizer = ({ onResize, property, headerText, messages }) => {
+const Resizer = ({ onResize, property, headerText, messages, headerId }) => {
   const { theme } = useThemeValue();
   const [active, setActive] = useState(false);
   const [start, setStart] = useState();
   const [width, setWidth] = useState();
   const ref = useRef();
   const { format } = useContext(MessageContext);
 
+  useEffect(() => {
```

Comment:
Could you explain why this is needed?

Reply:
Pushed up a commit that adds a comment to this useEffect

Reply:
Thanks !